app/routes/ffxiv.shortage-predictor.tsx

```
@@ -0,0 +1,301 @@
+              inputMap[field as keyof ShortagePredictorProps] ||
+              'Unknown input error'
+          )
+        )
+        .join(', ')}`
+    })
+  }
+
+  return FFXIVShortagePredictor(validInput.data)
+}
+
+type ActionResponse = PredictionResponse | { exception: string } | {}
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)